orientdb/src/main/scala/akka/stream/alpakka/orientdb/OrientDBFlowStage.scala

```diff
@@ -0,0 +1,209 @@
+      }
+
+      private def sendOSQLBulkInsertRequest(messages: Seq[IncomingMessage[T]]): Unit = {
+        val client = settings.oDatabasePool.acquire()
+        val oObjectClient = new OObjectDatabaseTx(client)
+
+        try {
+          if (clazz.isEmpty) {
+            if (!client.getMetadata.getSchema.existsClass(className)) {
+              client.getMetadata.getSchema.createClass(className)
+            }
+            client.begin(OTransaction.TXTYPE.OPTIMISTIC)
+
+            var faultyMessages: List[IncomingMessage[T]] = List()
+            messages.map(_.oDocument).foreach {
+              case oDocument: ODocument =>
+                val document = new ODocument()
+                oDocument
+                  .asInstanceOf[ODocument]
+                  .fieldNames()
+                  .zip(oDocument.asInstanceOf[ODocument].fieldValues())
+                  .foreach {
+                    case (fieldName, fieldVal) =>
+                      document.field(fieldName, fieldVal)
+                      ()
+                  }
+                document.setClassName(className)
+                client.save(document)
+                ()
+              case oRecord: ORecord =>
+                client.save(oRecord)
+                ()
+              case others: AnyRef =>
+                faultyMessages = faultyMessages ++ List(IncomingMessage[T](others.asInstanceOf[T]))
+                ()
+            }
+            client.commit()
+            if (faultyMessages.nonEmpty) {
+              responseHandler.invoke((faultyMessages, Some("Records are invalid OrientDB Records")))
+            } else {
+              responseHandler.invoke((Seq(), None))
+            }
+          } else {
+            oObjectClient.getEntityManager.registerEntityClass(
+              clazz.getOrElse(throw new RuntimeException("Typed stream class is invalid"))
+            )
+
+            var faultyMessages: List[IncomingMessage[T]] = List()
+            messages.map(_.oDocument).foreach {
+              case typeRecord: T =>
+                oObjectClient.save(typeRecord)
+                ()
+              case others: AnyRef =>
+                faultyMessages = faultyMessages ++ List(IncomingMessage[T](others.asInstanceOf[T]))
+                ()
+            }
+
+            if (faultyMessages.nonEmpty) {
+              responseHandler.invoke((faultyMessages, Some("Records are invalid OrientDB Records")))
+            } else {
+              responseHandler.invoke((Seq(), None))
+            }
+          }
+        } catch {
+          case exception: Exception =>
+            failureHandler.invoke((messages, exception))
+        } finally {
+          oObjectClient.close()
+          client.close()
+        }
+      }
+
+      setHandlers(in, out, this)
+
+      override def onPull(): Unit = tryPull()
+
+      override def onPush(): Unit = {
+        val message = grab(in)
+        queue.enqueue(message)
+
+        state match {
```

Comment:
Do you really need this state machine?
Can you simply call push after the previous insertion is done?